lib/utils/logDeprecation.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+const chalk = require('chalk');
+const memoizee = require('memoizee');
+
+const disabledCodesByEnv = extractCodes(process.env.SLS_DEPRECATION_DISABLE);
+
+const loggedDeprecations = new Set();
+
+function extractCodes(codesStr) {
+  if (!codesStr) {
+    return new Set();
+  }
+  return new Set(codesStr.split(','));
+}
+
+const resolveDeprecatedByService = memoizee((serviceConfig = {}) => {
```

Comment:
In this case `serviceConfig` should be treated as required argument (no default).

Note that configured as above, memoization will ignore `serviceConfig` argument, and will memoize function as if takes no arguments (only first invocation will be a cache miss, all others no matter what service config is passed will result with cache hit).

Reply:
This is handled by using `memoizee/weak`, right?

Reply:
When I try this simple demo
```javascript
const memoize = require('memoizee');
const func = memoize(function (obj) { return Object.keys(obj); });

console.log(func({e: 1}))
console.log(func({y: 1}))
```

This will print `[ 'e' ]` & `[ 'y' ]` respectively

Reply:
But this demo doesn't reflect what you've configured. Try below instead :)

```javascript
const memoize = require('memoizee');
const func = memoize(function (obj = {}) { return Object.keys(obj); });

console.log(func({e: 1}))
console.log(func({y: 1}))
```